### PR TITLE
feat: new admin settings profile page (step 1 of admin interface)

### LINKED ISSUE #254: Prepare a container for the administration interface and move the profile's settings into there
Mockups: [Figma](https://www.figma.com/design/kQOH4s5DnfWL0p6o6tlMQz/Draft-%5BATS%5D?node-id=1720-10529&t=Aivif5wlJq7UHGM0-1)

- [x] Container:
  - [x] Use Tabler’s [Settings](https://preview.tabler.io/settings.html) container and layout
  - [x] Keep our page container paddings (24px top&side, 48px bottom)
  - [x] Use our H1 for page heading (24px font-size, 32px line-height)
  - [x] By default page content should occupy half of the space on md breakpoint and larger
  - [x] Remove footer with buttons, instead place “Save” button below page content
    - No need for “Cancel” button
    - On success display full-width success alert above container with text “All changes have been saved.”
- [x] In user’s avatar dropdown menu, on clicking “Settings” option redirect to this interface on “Profile” page (see below)
- [x] For admins set links in navigation sidebar to this:
  - [x] Personal (title)
    - Profile
  - [x] Company (title)
    - General
  - [x] Recruitment (title)
    - Sources
    - Disqualify reasons
- [x] For members set links in navigation sidebar to this:
  - [x] Personal (title)
    - Profile
- [x] On mobile:
  - [x] Replace Tabler’s navigation sidebar with our tab button dropdown; If it takes 4 hours or more to implement or requires hacks, keep Tabler's version.
  - [x] Nice-to-have: if implemented as tab button dropdown, in dropdown menu add dropdown headers (for titles, add icons too) as in [Tabler](https://preview.tabler.io/dropdowns.html)
- [x] Add icons for titles in navigation sidebar:
  - [x] Personal: [user](https://tabler.io/icons/icon/user) (15x15px, outline,  stroke width 1.25, gray-500 color)
  - [x] Company: [building](https://tabler.io/icons/icon/building) (15x15px, outline, stroke width 1.25, gray-500 color)
  - [x] Recruitment: [search](https://tabler.io/icons/icon/search) (15x15px, outline, stroke width 1.25, gray-500 color)
- [x] Profile page
  - [x] Route: /settings/personal/profile
  - [x] Title: Settings - Personal - Profile | FreeATS
  - [x] Copy all content from “Settings” page
    - [x] Use tabler’s default font-weight for labels - 500
    - [x] Change heading to “Profile”
    - [x] Add “Enter full name” placeholder for Full name input
    - [x] Add “Password” label before “Change password” link
    - [x] Remove full-width param for “Save” button
    - [x] Set spacing as on mockup
  - [x] Remove “Settings” page
  - [x] Avatar upload, deletion and linking email should apply effect without need to press “Save” button as currently implemented

### PR DESCRIPTION
**Pull request by Lyams**
Closes #254.

- [x] I have reviewed my code in "Files changed" tab.
- [x] I have re-read the issue and this PR fully resolves it.

### CODE REVIEW

File: app/views/settings/shared/_tabs.html.slim [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+              data: { bs_toggle: :dropdown },
+            ) do |c|
+      - c.with_icon(:chevron_down)
+      span = tab_labels[active_tab]
+    ul.dropdown-menu.w-100 aria-labelledby="dropdownNavLink"
+      h6.dropdown-header
+        = inline_svg_tag("user.svg", { class: "me-2", style: "--tblr-icon-size: 15px" })
+        = t("user_accounts.personal")
+      li
+        = link_to t("user_accounts.profile"), settings_personal_profile_path,
```
Comment (wrmk) [inline app/views/settings/shared/_tabs.html.slim:30]: lets use `tab_labels[:profile]` to get locale, and similar logic below

File: app/views/settings/shared/_tabs.html.slim [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+              flex_content_position: :between,
+              id: "dropdownNavLink",
+              class: "w-100",
+              data: { bs_toggle: :dropdown },
+            ) do |c|
+      - c.with_icon(:chevron_down)
+      span = tab_labels[active_tab]
+    ul.dropdown-menu.w-100 aria-labelledby="dropdownNavLink"
+      h6.dropdown-header
+        = inline_svg_tag("user.svg", { class: "me-2", style: "--tblr-icon-size: 15px" })
```
Comment (wrmk) [inline app/views/settings/shared/_tabs.html.slim:27]: need to discuss icons after answer from Amgalan

File: app/views/settings/shared/_tabs.html.slim
Comment (wrmk) [inline app/views/settings/shared/_tabs.html.slim:1]: this file is very complex, but have no idea how to make it simpler quickly.

Comment (wrmk) [reply app/views/settings/shared/_tabs.html.slim:1]: btw, let's hide not implemented tabs to not confuse users

File: app/controllers/settings/personal/profiles_controller.rb [hunk 3]
```diff
@@ -62,29 +63,35 @@ def link_gmail
...
-      redirect_to ats_settings_path, notice: "Gmail successfully linked."
+      redirect_to settings_personal_profile_path, notice: "Gmail successfully linked."
     end
   end
 
   private
 
   def set_partial_variables
-    return unless allowed_to?(:link_gmail?, with: ATS::ProfilePolicy)
+    return unless allowed_to?(:link_gmail?, with: Settings::Personal::ProfilePolicy)
```
Comment (wrmk) [inline app/controllers/settings/personal/profiles_controller.rb:88]: btw, is this the same?
`allowed_to?(:link_gmail?, :profile)`

**Comment by wrmk** [comment]
on mobile screen dropdown has too much width
image

**Comment by wrmk** [comment]
please Add “Enter full name” placeholder for Full name input

File: app/assets/images/user.svg
[the remark this answers is not in the document]
Comment (amgalanbadmaev) [reply app/assets/images/user.svg:1]: Yes, links in description:

> Personal: [user](https://tabler.io/icons/icon/user) (15x15px, outline, stroke width 1.25, gray-500 color)
>  Company: [building](https://tabler.io/icons/icon/building) (15x15px, outline, stroke width 1.25, gray-500 color)
>  Recruitment: [search](https://tabler.io/icons/icon/search) (15x15px, outline, stroke width 1.25, gray-500 color)

**Comment by Lyams** [comment]
> on mobile screen dropdown has too much width image

By trial and error I reduced the dropdown so that it was not bigger than the button. Maybe that will be enough. To make the dropdown the size of the button, I do not see any simple options for implementing this without  JS.
Updated in 33bdf69af3e32b4476077f96d46b4ec6ebfe19ee.

File: app/assets/stylesheets/src/ats/_main.scss [hunk 4]
```diff
@@ -42,3 +42,8 @@ tr {
   padding-top: 0;
   padding-bottom: 0;
 }
+
+#settings .list-group-item {
+  padding-top: 16px;
+  padding-bottom: 16px;
+}
```
[the remark this answers is not in the document]
Comment (Lyams) [reply app/assets/stylesheets/src/ats/_main.scss:48]: Now I have replaced the custom css class with py-3.
But one or another solution is needed.

Vertical paddings without py-3/custom class are 12px:
Screenshot from 2024-11-27 11-14-49

On figma mockup vertical paddings are 16 px:
Screenshot from 2024-11-27 12-15-28

Vertical paddings with py-3/custom class are 16px:
Screenshot from 2024-11-27 11-15-16


Uses p-3 in 088c451c6d7956d1d29d34ea53fae20a2fd7e3ba.

File: app/assets/images/user.svg
[the remark this answers is not in the document]
Comment (Lyams) [reply app/assets/images/user.svg:1]: Fixed in d27d0dd0330ed93f8fe2addc8fef87eeab741e6c

File: app/controllers/settings/company/generals_controller.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class Settings::Company::GeneralsController < AuthorizedController
```
[the remark this answers is not in the document]
Comment (Lyams) [reply app/controllers/settings/company/generals_controller.rb]: Used GeneralProfilesController in e88c53851fe6ade105f47a69e99721f2ceca40f5

File: app/controllers/settings/company/generals_controller.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  layout "ats/application"
+
+  before_action { authorize! with: Settings::Company::GeneralPolicy }
```
[the remark this answers is not in the document]
Comment (Lyams) [reply app/controllers/settings/company/generals_controller.rb]: Fixed in 128c49987b00fe6639e98f28c33a6bfbc0225fcb.

File: app/views/settings/shared/_tabs.html.slim (hunk 1 above)
Comment (Lyams) [reply app/views/settings/shared/_tabs.html.slim]: Used in ce8d4e8791ddc0b8155b33ac0f537eb159cfd830

File: app/views/settings/shared/_tabs.html.slim (hunk 2 above)
Comment (Lyams) [reply app/views/settings/shared/_tabs.html.slim]: Fixed in d27d0dd0330ed93f8fe2addc8fef87eeab741e6c

**Comment by Lyams** [comment]
> please update tab name to `Settings - Personal - Profile | FreeATS` from this image

Fixed in f518be5ab7608ff4c3e5ef36bf6e7dc93086d478

**Comment by Lyams** [comment]
> please Add “Enter full name” placeholder for Full name input
Fixed in e1e0fcb066ae9d8354fbcef48f56b6ca0b609897.

**Comment by wrmk** [comment]
> By trial and error I reduced the dropdown so that it was not bigger than the button. Maybe that will be enough. To make the dropdown the size of the button, I do not see any simple options for implementing this without JS. Updated in [33bdf69](https://github.com/freeats/freeats/commit/33bdf69af3e32b4476077f96d46b4ec6ebfe19ee).

I pushed commit, please look at this

**Review by wrmk — COMMENTED** [review]
may i ask you to detect where is the problem?
navbar became mobile, but setting page not
image

**Comment by Lyams** [comment]
> may i ask you to detect where is the problem? navbar became mobile, but setting page not image

@amgalanbadmaev  Below what breakpoint should we switch to mobile display? Below lg or md?
I did it for the admin interface - below md. But the navbar changes to mobile when the screen is below lg. Is this correct?

File: config/locales/en.yml [hunk 7]
```diff
@@ -291,25 +310,8 @@ en:
     invite: Invite member
     joined_on: Joined on
     not_found: Account not found
-    personal: Personal
-    profile: Profile
```
[the remark this answers is not in the document]
Comment (Lyams) [reply config/locales/en.yml]: It has been added in this PR.

**Comment by amgalanbadmaev** [comment]
> > may i ask you to detect where is the problem? navbar became mobile, but setting page not image
> 
> @amgalanbadmaev Below what breakpoint should we switch to mobile display? Below lg or md? I did it for the admin interface - below md. But the navbar changes to mobile when the screen is below lg. Is this correct?

Should be always below md.

**Comment by wrmk** [comment]
@Lyams may i ask you to create the issue about navbar.
ask me for review if you're ok with my commit

**Comment by Lyams** [comment]
> Should be always below md.

Created issue https://github.com/freeats/freeats/issues/269

**Pull request merged**